### ISSUE TITLE
Unsize rules issues

### ISSUE THREAD

**Issue by compiler-errors** [rust-lang/fls#457]
https://github.com/ferrocene/specification/blob/496ef38bcfe074d88580021b5b3fadc0239c85b2/src/types-and-traits.rst?plain=1#L2011-L2012

This rule is not required. [This code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ecef743eaca1279b8b5c1c633af6cb35) compiles even though `i32: Unsize<u32>` is *not* true.

https://github.com/ferrocene/specification/blob/496ef38bcfe074d88580021b5b3fadc0239c85b2/src/types-and-traits.rst?plain=1#L2017-L2020

This rule should be generalized to allow multiple parameters to be changed. See the example committed in rust-lang/rust#117495 (`tests/ui/unsized/unsize-coerce-multiple-adt-params.rs`).

Also, [this `X` field](https://github.com/ferrocene/specification/blob/496ef38bcfe074d88580021b5b3fadc0239c85b2/src/types-and-traits.rst?plain=1#L2002) isn't referenced later in the document.

**Comment by PLeVasseur** [comment]
@traviscross -- could you provide some guidance on which `A-` label is most appropriate in your mind?

Maybe `A-types`?

**Comment by traviscross** [comment]
Yes, reasonable.  It'd probably get `A-coercions" as well on `r-l/r`.  I've added that label.
